inc/aiosp_common.php

```diff
@@ -182,4 +189,256 @@ public static function make_xml_safe( $tag, $value ) {
 		return $value;
 	}
 
+	/**
+	 * Attachment URL to Post ID
+	 *
+	 * Returns the (original) post/attachment ID from the URL param given. The function checks if URL is
+	 * within, chacks for original attachment URLs, and then custom attachment URLs. The main intent for this function
+	 * is to avoid having to query if possible (if cache was set prior), and if not, there is only 1 query per instance
+	 * rather than multiple queries per instance.
+	 * NOTE: Attempting to paginate the query actually caused the memory to peak higher.
+	 * NOTE: The weakest point in this function is multiple calls to Result_2's SQL query for custom attachment URLs.
+	 *
+	 * This is intended to work much the same way as WP's `attachment_url_to_postid()`.
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/attachment_url_to_postid/
+	 *
+	 * @see aiosp_common::set_transient_url_postids()
+	 * @see get_transient()
+	 * @link https://developer.wordpress.org/reference/functions/get_transient/
+	 * @see wpdb::get_results()
+	 * @link https://developer.wordpress.org/reference/classes/wpdb/get_results/
+	 * @see wp_list_pluck()
+	 * @link https://developer.wordpress.org/reference/functions/wp_list_pluck/
+	 * @see wp_upload_dir()
+	 * @link https://developer.wordpress.org/reference/functions/wp_upload_dir/
+	 *
+	 * @since 2.9.2
+	 *
+	 * @param string $url Full image URL.
+	 * @return int
+	 */
+	public static function attachment_url_to_postid( $url ) {
+		global $wpdb;
+		static $results_1;
+		static $results_2;
+
+		$id = 0;
+		$url_md5 = md5( $url );
+
+		// Gets the URL => PostIDs array.
+		// If static variable is still empty, load transient data.
+		if ( is_null( self::$attachment_url_postids ) ) {
+			if ( is_multisite() ) {
+				self::$attachment_url_postids = get_site_transient( 'aioseop_multisite_attachment_url_postids' );
+			} else {
+				self::$attachment_url_postids = get_transient( 'aioseop_attachment_url_postids' );
+			}
+
+			// If no transient data, set as (default) empty array.
+			if ( false === self::$attachment_url_postids ) {
+				self::$attachment_url_postids = array();
+			}
+		}
+
+		// Search for URL and get ID.
+		if ( isset( self::$attachment_url_postids[ $url_md5 ] ) ) {
+			// If static is already loaded and has URL, then return the URL's Post ID.
+			$id = intval( self::$attachment_url_postids[ $url_md5 ] );
+		} else {
+			// Check to make sure Image URL is not outside the website.
+			$uploads_dir = wp_upload_dir();
```

Comment:
@EkoJR better to use wp_get_upload_dir as its lightweight https://developer.wordpress.org/reference/functions/wp_get_upload_dir/

Reply:
I thought the minimum version was 4.5 but I'm not sure if it makes sense to put this code with a version check.

Reply:
My thoughts exactly. So, either would need to...

* Increase the WP version requirement from 4.4 to 4.5.
* Keep as wp_upload_dir.

@michaeltorbert Should we increase the required version?